docs/metrics/extending-the-sdk/MyReader.cs

```diff
@@ -29,7 +30,18 @@ public MyReader(string name = "MyReader")
 
     protected override bool ProcessMetrics(Batch<Metric> metrics, int timeoutMilliseconds)
     {
-        Console.WriteLine($"{this.name}.ProcessMetrics(metrics={metrics}, timeoutMilliseconds={timeoutMilliseconds})");
+        var sb = new StringBuilder();
+        foreach (var record in metrics)
+        {
+            if (sb.Length > 0)
+            {
+                sb.Append(", ");
+            }
+
+            sb.Append($"{record}");
+        }
+
+        Console.WriteLine($"{this.name}.ProcessMetrics(metrics=[{sb}], timeoutMilliseconds={timeoutMilliseconds})");
```

Comment:
nit: If we care about readability we could simply use `string.Join`. If we care about performance we could use some `bool` variable instead of checking `sb.Length > 0` each time.

Reply:
Good point!

I think here we care about readability, not perf.
I guess we might need to go through each metric and visit all the points, so `string.Join` probably won't work well here? (e.g. this is what we have for logs https://github.com/open-telemetry/opentelemetry-dotnet/blob/f05b05861397adfb38f5425dd3cc8c2f42a2ac21/docs/logs/extending-the-sdk/MyExporter.cs#L38)

Given we don't have such logic right now, I'll simply the code in this PR.

Reply:
Just tried and I couldn't figure out a simpler way 🤷.

`string.Join(", ", batch);` would not iterate through `batch`, and if we try to cast it / convert it, the result is actually less readable (e.g. with LINQ).

Reply:
I was thinking about something like:

```csharp
var metricsDump = string.Join(", ", metrics.Select(x => x.ToString()).ToArray());
Console.WriteLine($"{this.name}.ProcessMetrics(metrics=[{metricsDump}], timeoutMilliseconds={timeoutMilliseconds})");
```

what is more readable? I have no idea :smile:

feel free to resolve the comment and leave it as it is